app/cdap/components/Metadata/Lineage/LineageDiagram/index.tsx

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { Redirect, useHistory, useParams, useLocation } from 'react-router-dom';
+import 'jsplumb';
+import T from 'i18n-react';
+import styled from 'styled-components';
+import { MySearchApi } from 'api/search';
+import { getScale } from 'components/Metadata/Lineage/helper';
+import uuidV4 from 'uuid/v4';
+
+const I18N_PREFIX = 'features.MetadataLineage';
+const NODE_HEIGHT = '55px';
+const NODE_WIDTH = '175px';
+const DAG_CONTAINER_ID = `dag-${uuidV4()}`;
+
+const NoContentMsg = styled.p`
+  margin-top: 150px;
+  text-align: center;
+  font-size: 1.39em;
+  color: #666;
+`;
+
+const LineageContainer = styled.div`
```

Comment:
Can you split the child styles up into their own styled components?

Reply:
@njbriggs ignore this PR for now.